R/join_keys.R

```diff
@@ -142,6 +142,7 @@ join_keys.teal_data <- function(...) {
 #' join_keys(td)
 `join_keys<-.teal_data` <- function(x, value) {
   join_keys(x@join_keys) <- value
+  x@datanames <- sort_datanames(x@datanames, x@join_keys)
```

Comment:
reload datanames

```suggestion
  datanames(x) <- x@datanames
```

Reply:
really nice, but needed to think a couple of time what it does.
If we have `x@datanames <- sort_datanames(x@datanames, x@join_keys)` then you know sorting is done.
If we have `datanames(x) <- x@datanames` then only if you know behavior `datanames(x)` you know that sorting is applied.

So maybe, to make it clear, let's have a comment:
```r
datanames(x) <- x@datanames  #datanames() triggers sort_datanames()/topological_sort()
```

Reply:
There is one thing missing. If we want to set datanames and include parent-datanames we need to `setValidity` on `@datanames` ~ `names(@env)` consistency